class PersonError extends Error {
    constructor(message) {
        super(message);
        this.name = 'PersonError';
    }
}

class HttpError extends Error {
    constructor(code, status) {
        super(status);
        this.name = 'HttpError';
        this.status = status;
        this.code = code;
    }
}


const person = {
    name: 'Bill'
}

// function getPersonInfo() {
//     if(!person['age']) {
//         throw new PersonError('Person has not age'); // Это должна быть ошибка 500
//     }
//     return person[age];
// }

function getPage(url) {
    if(url !== 'test.html') {
        throw new HttpError(404, 'Not found page ' + url); // Это должна быть ошибка 404
    }
}

try {
    // getPersonInfo();
    getPage('index.html');
}catch(e) {
    if(e instanceof HttpError) {
        console.log(e.code, e.message)
    } else {
        console.log(`Error message: ` + e.message)
    }
}
